assign_nodes.py

```diff
@@ -24,40 +26,91 @@ def unique_nodes(examples):
 	return(results)
 
 # Create a list of nodes for input
-def search_nodes(nodes, kg):
-	user_nodes = []
+
+# Could potentially find several features for a single input example. Need a way to be able to select multiple feaures for a search. 
+# Need a way to go back through search terms. 
+
+def search_nodes(nodes, kg, examples):
+	examples["source_label"] = ""
+	examples["target_label"] = ""
 	for node in nodes:
 		print("User Search Node: ", node)
-		found_nodes = find_node(node,kg)
-		
+		found_nodes = find_node(node,kg)		
 		nrow = found_nodes.shape[0]
+		print("Found", nrow, "features in KG")
+		user_input = ""
+		bad_input = True
 		if nrow < 20:
-			print(found_nodes.iloc[0:20,])
-			user_input = input("Input node integer_id: ")
-		else:
-			n = range(0, nrow, 20)
-			user_input = ""
-			i = 0
-			while(type(user_input) != int):
-				print(found_nodes.iloc[n[i]:n[i+1],])
-				user_input = input("Input node integer_id or hit enter for the next 20 features: ")
+			while(bad_input):
+				print(found_nodes.iloc[0:nrow,])
+				user_input = input("Input node integer_id: ")
 				try:
 					user_input = int(user_input)
 				except ValueError:
-					i+=1
-		user_nodes.append(user_input)
-	return(user_nodes)
+					print("Input not an input_id.... try again")
+				else:
+					if node_in_search(found_nodes,user_input):
+						print("Input is an input_id")
+						node_label = found_nodes[found_nodes["integer_id"] == user_input]["label"].item()
+						bad_input = False
+					else:
+						print("Input not an input_id.... try again")
+		else:	
+			i = 0
+			while(bad_input):
+				high = min(nrow,(i+1)*20)
+				print(found_nodes.iloc[i*20:high,])
+				user_input = input("Input node integer_id or 'f' for the next 20 features or 'b' for the previous 20: ")
+				try:
+					user_input = int(user_input)
+				except:
+					if user_input == 'f':
+						i+=1
+					elif user_input == 'b':
+						i-=1
+					else:
+						i+=1
+				else:
+					if node_in_search(found_nodes,user_input):
+						print("Input is an input_id")
+						node_label = found_nodes[found_nodes["integer_id"] == user_input]["label"].item()
+						bad_input = False
+					else:
+						print("Input not an input_id.... try again")
+		examples.loc[examples["source"] == node,"source_label"] = node_label
+		examples.loc[examples["target"] == node,"target_label"] = node_label
+	return(examples)
+
 
-	
+<<<<<<< HEAD
+# Check if search input is in the list of integer_ids
+def node_in_search(found_nodes, user_input):
+	if user_input in list(found_nodes["integer_id"]):
+		return(True)
+	else:
+		return(False)
 
+#subgraph_df is a dataframe with source,targe headers and | delimited
+def create_input_file(examples,output_dir):
 
+    input_file = output_dir+"/_Input_Nodes_.csv"
```

Comment:
A naming idea is Indexed_Input_Nodes or KG_Mapped_Input_Nodes for this file, which will be fed into the subgraph steps in the create_subgraph.py

Reply:
yes, will add to issues